source/unplugin/unplugin.civet

```diff
@@ -52,23 +52,30 @@ type CacheEntry
   result?: TransformResult
   promise?: Promise<void>
 
-civetExtension := /\.civet$/
+civetExtension := /\\.civet$/
 // Normally .jsx/.tsx extension should be present, but sometimes
 // (e.g. esbuild's alias feature) loads directly without resolve
-isCivetTranspiled := /(\.civet)(\.[jt]sx)?([?#].*)?$/
 postfixRE := /[?#].*$/s
 isWindows := os.platform() is 'win32'
 windowsSlashRE := /\\/g
-
-// Extracts query string and hash, and removes tsx/jsx extension
-function cleanCivetId(id: string): {id: string, postfix: string}
-  let postfix = ''
-  id = id
-  .replace postfixRE, (match) =>
+civetSuffix := '.civet'
+
+// Validate and extract the base Civet filename dynamically using outExt
+function validateAndExtractCivetBase(id: string, outputExtension: string): string | null
+  unless id? and outputExtension? return null
```

Comment:
```suggestion
  return unless id? and outputExtension?
```

Actually, I think this line can safely be removed: typing should guarantee we never hit this condition. We don't need to guard against misuse errors given that we're the only callers.